src/sentry/models/event.py

```diff
@@ -470,15 +484,42 @@ def get_event(cls, project_id, event_id):
         return None
 
     def __init__(self, snuba_values):
-        assert set(snuba_values.keys()) == set(self.selected_columns)
+        """
+            When initializing a SnubaEvent, think about the attributes you
+            might need to access on it. If you only need a few properties, and
+            they are all available in snuba, then you should use
+            `SnubaEvent.selected_colums` (or a subset depending on your needs)
+            But if you know you are going to need the entire event body anyway
+            (which requires a nodestore lookup) you may as well just initialize
+            the event with `SnubaEvent.minimal_colums` and let the rest of of
+            the attributes come from nodestore.
+        """
+        assert all(k in snuba_values for k in SnubaEvent.minimal_columns)
 
-        self.__dict__ = snuba_values
+        # self.snuba_data is a dict of all the stuff we got from snuba
+        self.snuba_data = snuba_values
 
-        # This should be lazy loaded and will only be accessed if we access any
-        # properties on self.data
-        node_id = SnubaEvent.generate_node_id(self.project_id, self.event_id)
+        # self.data is a (lazy) dict of everything we got from nodestore
+        node_id = SnubaEvent.generate_node_id(
+            self.snuba_data['project_id'],
+            self.snuba_data['event_id'])
         self.data = NodeData(None, node_id, data=None)
 
+    def __getattr__(self, name):
```

Comment:
This is instead of just setting `self.__dict__` to the snuba result, which seemed a little janky.